Source/WebCore/platform/network/soup/CertificateInfoSoup.cpp

```diff
@@ -124,20 +125,16 @@ std::optional<CertificateSummary> CertificateInfo::summary() const
     if (subjectName)
         summaryInfo.subject = String::fromUTF8(subjectName.get());
     if (dnsNames) {
-        for (unsigned i = 0; i < dnsNames->len; ++i) {
-            WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GLib port
-            GBytes* bytes = static_cast<GBytes*>(dnsNames->pdata[i]);
-            WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+        for (auto dnsName : span(dnsNames)) {
+            GBytes* bytes = static_cast<GBytes*>(dnsName);
             gsize dataLength;
             const auto* data = g_bytes_get_data(bytes, &dataLength);
             summaryInfo.dnsNames.append(String({ static_cast<const char*>(data), dataLength }));
         }
     }
     if (ipAddresses) {
-        for (unsigned i = 0; i < ipAddresses->len; ++i) {
-            WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GLib port
-            GUniquePtr<char> ipAddress(g_inet_address_to_string(static_cast<GInetAddress*>(ipAddresses->pdata[i])));
-            WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+        for (auto address : span(ipAddresses)) {
+            GUniquePtr<char> ipAddress(g_inet_address_to_string(static_cast<GInetAddress*>(address)));
```

Comment:
Here we can pass the type when building the `std::span` as well:

```suggestion
        for (auto* address : span<GInetAddress*>(ipAddresses)) {
            GUniquePtr<char> ipAddress(g_inet_address_to_string(address));
```